tests/test_specs.py

```diff
@@ -457,3 +457,66 @@ async def test_openai_embedding_spec_with_batching(openai_embedding_request_data
             assert len(resp2.json()["data"]) == 4, "Length of data should be 4"
             assert len(resp1.json()["data"][0]["embedding"]) == 768, "Embedding length should be 768"
             assert len(resp2.json()["data"][0]["embedding"]) == 768, "Embedding length should be 768"
+
+
+class IncorrectAsyncAPI1(ls.LitAPI):
+    def setup(self, device):
+        self.model = None
+
+    async def decode_request(self, request, **kwargs):
+        return request
+
+    async def predict(self, x):
+        return "This is a generated output"
+
+    async def encode_response(self, output):
+        return ChatMessage(role="assistant", content="This is a generated output")
+
+
+class IncorrectAsyncAPI2(IncorrectAsyncAPI1):
+    async def predict(self, x):
+        yield "This is a generated output"
+
+
+@pytest.mark.asyncio
+async def test_openai_spec_with_async_lit_api_validation(openai_request_data):
+    with pytest.raises(ValueError, match="predict is not a generator"):
+        ls.LitServer(IncorrectAsyncAPI1(enable_async=True), spec=OpenAISpec())
+
+    with pytest.raises(ValueError, match="encode_response is not a generator"):
+        ls.LitServer(IncorrectAsyncAPI2(enable_async=True), spec=OpenAISpec())
+
+    with pytest.raises(ValueError, match="'enable_async' is not set in LitAPI."):
+        ls.LitServer(IncorrectAsyncAPI2(enable_async=False), spec=OpenAISpec())
+
+
+class AsyncOpenAILitAPI(ls.LitAPI):
+    def setup(self, device):
+        self.model = None
+        self.sentence = ["This", " is", " a", " sample", " response"]
+
+    async def decode_request(self, request, **kwargs):
+        return request
+
+    async def predict(self, x):
+        for token in self.sentence:
+            yield token
+
+    async def encode_response(self, output_stream, context):
+        for output in output_stream:
+            yield {"role": "assistant", "content": output}
+
```

Comment:
let's also test scenario when decode is not implemented by the user

Reply:
sure. done